src/Microsoft.Health.Dicom.Api/Features/Context/DicomRequestContextMiddleware.cs

```diff
@@ -51,8 +52,12 @@ public async Task Invoke(HttpContext context, IDicomRequestContextAccessor dicom
 
             dicomRequestContextAccessor.RequestContext = dicomRequestContext;
 
-            // Call the next delegate/middleware in the pipeline
-            await _next(context);
+            // Adding operation id for this request thread to the log context.
+            using (LogContext.PushProperty("operationId", System.Diagnostics.Activity.Current?.RootId))
```

Comment:
Did'nt your previous change already add this to the DicomRequestContext to show up in Exceptions. How come it shows up in Exceptions and not in Logs?

Reply:
That was added just for the audit logs which are different than usual trace logs.

Reply:
For clarification, change in https://github.com/microsoft/dicom-server/pull/794 puts correlation Id for logs and exception logs that end up in Audit Logs (IFx and Shoebox). They follow a separate path than this. This information, however, was missing for trace logs that end up in KubernetesContainers table which this pull request should fix.

Reply:
It still does not make sense. If the linked PR was for audit logs how did it fix LinuxExceptionTelemetry? I though Audit logs are different. 
I would think Exceptions and normal logs would go through same pipeline.

Reply:
It did not :)
It fixed the audit logs, i.e. the IFx Audit Logs and the Azure Monitor Shoebox Logs. The reason I mentioned LinuxRequestTelemetry and LinuxExceptionTelemetry tables in the previous PR is that now customers will see CorrelationId in their logs i.e. the Audit Logs. When customers provide us with the correlation id, we can use that correlation id and get more details from LinuxRequestTelemetry and LinuxExceptionTelemetry table. So the previous PR had the change for both exceptions and normal audit logs.
This PR is focused on adding correlation id to the trace logs that end up in KubernetesContainers table. After this PR, we can use the same correlation id provided by the user and get further information using the trace logs in KubernetesContainers table.

Reply:
LinuxExceptionTelemetry does not even have Operation_id. How will be relate it? Was'nt the previous bug to allow joining of Request and Exception telemetry? How can we join if the Operation_id is not populated

Reply:
Actually LinuxExceptionTelemetry does have operation_id. Probably what you saw is data from IoT and FHIR. Let's get on a call. I have set up some custom exceptions so I can present a small demo to clarify these things.